src/ui/server/controllers/quote/tell-us-about-your-policy/index.ts

```diff
@@ -93,95 +94,34 @@ const generatePageVariables = (policyType: string) => {
-  } else {
-    mappedCreditPeriod = mapCreditPeriod(creditPeriodOptions);
-  }
-
-  const PAGE_VARIABLES = generatePageVariables(submittedData.quoteEligibility[POLICY_TYPE]);
-
-  return res.render(TEMPLATE, {
-    ...PAGE_VARIABLES,
-    BACK_LINK: req.headers.referer,
-    isSinglePolicyType: isSinglePolicyType(submittedData.quoteEligibility[POLICY_TYPE]),
-    isMultiPolicyType: isMultiPolicyType(submittedData.quoteEligibility[POLICY_TYPE]),
-    currencies: mappedCurrencies,
-    percentageOfCover: mappedPercentageOfCover,
-    creditPeriod: mappedCreditPeriod,
-    submittedValues: submittedData.quoteEligibility,
-  });
-};
-
-const post = async (req: Request, res: Response) => {
-  const { submittedData } = req.session;
-
-  const validationErrors = generateValidationErrors({
-    ...submittedData.quoteEligibility,
-    ...req.body,
-  });
-
-  const currencies = await api.external.getCurrencies();
-
-  if (!currencies || !currencies.length) {
-    return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
-  }
-
-  const submittedCurrencyCode = req.body[FIELD_IDS.CURRENCY];
-
-  if (validationErrors) {
-    // map currencies drop down options
-    let mappedCurrencies = [];
+    if (!isPopulatedArray(currencies)) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
 
-    if (submittedCurrencyCode) {
-      mappedCurrencies = mapCurrencies(currencies, submittedCurrencyCode);
+    let mappedCurrencies;
+    if (submittedData && submittedData.quoteEligibility && submittedData.quoteEligibility[FIELD_IDS.CURRENCY]) {
+      mappedCurrencies = mapCurrencies(currencies, submittedData.quoteEligibility[FIELD_IDS.CURRENCY].isoCode);
     } else {
       mappedCurrencies = mapCurrencies(currencies);
     }
 
-    // map percentage of cover drop down options
-    let mappedPercentageOfCover = [];
-    const submittedPercentageOfCover = req.body[PERCENTAGE_OF_COVER];
+    let mappedPercentageOfCover;
 
-    if (submittedPercentageOfCover) {
-      mappedPercentageOfCover = mapPercentageOfCover(PERCENTAGES_OF_COVER, submittedPercentageOfCover);
+    if (submittedData && submittedData.quoteEligibility && submittedData.quoteEligibility[PERCENTAGE_OF_COVER]) {
+      mappedPercentageOfCover = mapPercentageOfCover(PERCENTAGES_OF_COVER, Number(submittedData.quoteEligibility[PERCENTAGE_OF_COVER]));
     } else {
       mappedPercentageOfCover = mapPercentageOfCover(PERCENTAGES_OF_COVER);
     }
 
-    // map credit period drop down options
-    let mappedCreditPeriod = [];
-    const submittedCreditPeriod = req.body[CREDIT_PERIOD];
-
     const creditPeriodOptions = FIELDS[FIELD_IDS.CREDIT_PERIOD].OPTIONS as Array<SelectOption>;
+    let mappedCreditPeriod;
 
-    if (submittedCreditPeriod) {
-      mappedCreditPeriod = mapCreditPeriod(creditPeriodOptions, submittedCreditPeriod);
+    if (submittedData && submittedData.quoteEligibility && submittedData.quoteEligibility[CREDIT_PERIOD]) {
```

Comment:
can we simplify this to:
`if ( submittedData?.quoteEligibility?.[CREDIT_PERIOD])`?
the `?` should check if the property exists at any point

Reply:
Zain left the same comment - would like to look into these things in the future, there's a few instances. These haven't been changed in this PR, just the indentation has been changed from adding a try/catch. Will create a ticket

Reply:
sounds good